File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/PressureMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Pressure
import com.phamtunglam.health_connector_hc_android.pigeon.PressureDto

/**
 * Converts a [PressureDto] to a Health Connect [Pressure] object.
 */
internal fun PressureDto.toHealthConnect(): Pressure =
    Pressure.millimetersOfMercury(millimetersOfMercury)

/**
 * Converts a Health Connect [Pressure] object to a [PressureDto].
 *
 * Uses millimeters of mercury as the transfer unit.
 */
internal fun Pressure.toDto(): PressureDto = PressureDto(
    millimetersOfMercury = inMillimetersOfMercury,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/pressure_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Pressure;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show PressureDto, PressureUnitDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Pressure] to [PressureDto].
@internal
extension PressureToDto on Pressure {
  PressureDto toDto() {
    // Uses millimeters of mercury as the transfer unit.
    return PressureDto(
      value: inMillimetersOfMercury,
      unit: PressureUnitDto.millimetersOfMercury,
    );
  }
}

/// Converts [PressureDto] to [Pressure].
@internal
extension PressureDtoToDomain on PressureDto {
  Pressure toDomain() {
    switch (unit) {
      case PressureUnitDto.millimetersOfMercury:
        return Pressure.millimetersOfMercury(value);
    }
  }
}
